File: snippets/ruby/get_the_results_of_a_batch_operation.rb
require "net/http"

response_body_url = URI.parse("RESPONSE_BODY_URL_FROM_PREVIOUS_STEPS")
response = Net::HTTP.get(response_body_url)

# Example data extraction from gzipped archive, returns as array of hashes
results = processBatchArchive(response)
results.each do |result|
  user = User.find(result[:operation_id])
  user.update mailchimp_web_id: result[:response][:web_id]
end

File: snippets/php/get_the_results_of_a_batch_operation.php
require dirname(__DIR__).'/vendor/autoload.php';

$url = "RESPONSE_BODY_URL_FROM_PREVIOUS_STEPS";
$guzzle = new GuzzleHttp\Client();
$response = $guzzle->get($url);

// Extract data from gzipped archive, return as array
// Implementation details not included
$results = processBatchArchive($response);

foreach ($results as $result)
{
    $user = FakeUser::find($result['operation_id']);
    $user->setMailchimpWebId(result['response']['web_id']);
    $user->save();
}